File: src/main/java/workshop/part2/FraudWordActor.java
package workshop.part2;

import akka.actor.AbstractActor;
import javaslang.collection.List;
import workshop.common.fraudwordsservice.FraudWord;
import workshop.common.fraudwordsservice.FraudWordService;

public class FraudWordActor extends AbstractActor {

    private final FraudWordService fraudWordService;

    public FraudWordActor(FraudWordService fraudWordService) {
        this.fraudWordService = fraudWordService;
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .build();
    }

    // Request message type
    public static class ExamineWords {
        public final List<String> words;

        public ExamineWords(List<String> words) {
            this.words = words;
        }
    }

    // Reply message type
    public static class ExamineWordsResult {
        public final List<FraudWord> fraudWords;

        public ExamineWordsResult(List<FraudWord> fraudWords) {
            this.fraudWords = fraudWords;
        }
    }
}

File: src/main/java/workshop/part2/FraudWordActor.kt
package workshop.part2

import akka.actor.UntypedActor
import workshop.common.fraudwordsservice.FraudWord
import workshop.common.fraudwordsservice.FraudWordService

class FraudWordActor(private val fraudWordService: FraudWordService) : UntypedActor() {

    override fun onReceive(msg: Any?) = when (msg) {
        is ExamineWords -> sender().tell(ExamineWordsResult(fraudWordService.examineWords(msg.words)), sender())
        else -> unhandled(msg)
    }

    class ExamineWords(val words: List<String>)

    class ExamineWordsResult(val fraudWords: List<FraudWord>)
}
